javaslang/generator/Generator.scala

```diff
@@ -1265,46 +1265,40 @@ def generateMainClasses(): Unit = {
 
             ${(i == 0).gen(xs"""
               @Override
-              public <T> Tuple1<T> prepend(T value) {
+              public <T> Tuple1<T> append(T value) {
                  return new Tuple1<>(value);
               }
 
               @Override
-              public <T> Tuple1<T> append(T value) {
+              public <T> Tuple1<T> prepend(T value) {
                  return new Tuple1<>(value);
               }
             """)}
 
             ${(i > 0 && i < N).gen(xs"""
               @Override
-              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
-                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
-              }
-
-              @Override
               public <T> Tuple${i + 1}<${(1 to i).gen(j => s"T$j")(", ")}, T> append(T value) {
                   return Tuple.of(${(1 to i).gen(j => s"_$j")(", ")}, value);
               }
-            """)}
 
-            ${(i == N).gen(xs"""
               @Override
-              public <T> Tuple prepend(T value) {
-                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
+              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
+                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
               }
+            """)}
 
+            ${(i == N).gen(xs"""
               @Override
               public <T> Tuple append(T value) {
                   throw new UnsupportedOperationException("Append to Tuple${N}");
               }
-            """)}
 
-            ${(i > 0).gen(xs"""
-              public static $generics $className<${(1 to i).gen(j => s"Seq<? extends T$j>")(", ")}> sequence(Iterable<$className<${(1 to i).gen(j => s"? extends T$j")(", ")}>> tuples) {
-                Objects.requireNonNull(tuples, "tuples is null");
-                return new $className<>(${(1 to i).gen(j => s"${im.getType("javaslang.collection.Iterator")}.ofAll(tuples).map($className::_$j).toList()")(", ")});
+              @Override
+              public <T> Tuple prepend(T value) {
+                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
```

Comment:
I agree, that's not safe.

In an early stage of Javaslang I experimented with the integration of Tuple into the Value hierarchy. There were several problems, for details see #692 (e.g. [this](https://github.com/javaslang/javaslang/issues/692#issuecomment-151115513)). In the end I understood: 'A Tuple is a Tuple'. Not more, not less. It isn't a Seq, nor a Value.

We should not change the return type - too much confusion / not straight forward enough. Nor we should throw - that's unsafe.

I think these are our options:

* remove `append()`, `prepend()` from the base interface `Tuple` and remove them from `Tuple8`
* remove these methods completely

I don't know if these methods are used often in practice, haven't seen a use-case yet. So we should consider to remove them completely. Keep things simple! Less is more.